contracts/test/upgrade-test.ts

```diff
@@ -347,4 +364,54 @@ describe("Upgrade", async function () {
     expect(await vg1.walletFromHash(hash2)).to.equal(wallet1.address);
     expect(await vg1.hashFromWallet(wallet1.address)).to.equal(hash2);
   });
+
+  it("should NOT allow walletAdminCall where first param is not calling wallet", async function () {
+    const wallet1 = await fx.lazyBlsWallets[0]();
+    const wallet2 = await fx.lazyBlsWallets[1]();
+
+    const txnReceipt = await proxyAdminCall(fx, wallet1, "upgrade", [
+      wallet2.address,
+      wallet2.address,
+    ]);
+    const opResults = getOperationResults(txnReceipt);
+
+    expect(opResults).to.have.lengthOf(1);
+    expect(opResults[0].error.actionIndex.toNumber()).to.eql(0);
+    expect(opResults[0].error.message).to.eql(
+      "VG: first param to proxy admin is not calling wallet",
+    );
+  });
+
+  it("should NOT allow walletAdminCall to ProxyAdmin.transferOwnership", async function () {
+    const wallet = await fx.lazyBlsWallets[0]();
+
+    const txnReceipt = await proxyAdminCall(fx, wallet, "transferOwnership", [
+      wallet.address,
+    ]);
+    const opResults = getOperationResults(txnReceipt);
+
+    expect(opResults).to.have.lengthOf(1);
+    expect(opResults[0].error.actionIndex.toNumber()).to.eql(0);
+    expect(opResults[0].error.message).to.eql("VG: cannot change ownership");
+  });
+
+  /**
+   * TODO Figure why renounceOwnership call failed without error message
+   * result comes back as ["0x"] and success as false, figure out why.
+   */
+  it.skip("should NOT allow walletAdminCall to ProxyAdmin.renounceOwnership", async function () {
+    const wallet = await fx.lazyBlsWallets[0]();
+
+    const txnReceipt = await proxyAdminCall(
+      fx,
+      wallet,
+      "renounceOwnership",
+      [],
+    );
+    const opResults = getOperationResults(txnReceipt);
+
+    expect(opResults).to.have.lengthOf(1);
+    expect(opResults[0].error.actionIndex.toNumber()).to.eql(0);
+    expect(opResults[0].error.message).to.eql("VG: cannot change ownership");
+  });
```

Comment:
Need to investigate and fix this

Reply:
Fixed: https://github.com/web3well/bls-wallet/pull/391